golem-common/src/client.rs

```diff
@@ -98,42 +111,60 @@ pub struct MultiTargetGrpcClient<T: Clone> {
     config: GrpcClientConfig,
     clients: Arc<DashMap<http_02::Uri, GrpcClientConnection<T>>>,
     client_factory: Arc<dyn Fn(Channel) -> T + Send + Sync>,
+    target_name: String,
 }
 
 impl<T: Clone> MultiTargetGrpcClient<T> {
     pub fn new(
+        target_name: impl AsRef<str>,
         client_factory: impl Fn(Channel) -> T + Send + Sync + 'static,
         config: GrpcClientConfig,
     ) -> Self {
         Self {
             config,
             clients: Arc::new(DashMap::new()),
             client_factory: Arc::new(client_factory),
+            target_name: target_name.as_ref().to_string(),
         }
     }
 
-    pub async fn call<F, R>(&self, endpoint: http_02::Uri, f: F) -> Result<R, Status>
+    pub async fn call<F, R>(
+        &self,
+        description: impl AsRef<str>,
+        endpoint: http_02::Uri,
+        f: F,
+    ) -> Result<R, Status>
     where
         F: for<'a> Fn(&'a mut T) -> Pin<Box<dyn Future<Output = Result<R, Status>> + 'a + Send>>
             + Send,
     {
         let mut retries = RetryState::new(&self.config.retries_on_unavailable);
+        let span = debug_span!(
+            "gRPC call",
+            target_name = self.target_name,
+            endpoint = endpoint.to_string(),
+            description = description.as_ref()
+        );
+        let _span = span.enter();
```

Comment:
i think this will create async span leak problems, AFAIU:
 - in_current_span should be used with spawn, not in general async code
 - in code which uses await no span should be entered, as that will leak to other random tasks
 - rather the span should be only entered for sync blocks (and be dropped before using await) and use instrument for async ones

Reply:
Thanks, forgot that it works like that